service-workers/service-worker/registration-updateviacache.https.html

```diff
@@ -0,0 +1,230 @@
+
+      if (reg) {
+        frame.parentNode.removeChild(frame);
+        return reg;
+      }
+
+      await wait(100);
+    }
+  }
+
+  const registrationMethods = [
+    [registerViaApi, 'via-api'],
+    [registerViaLinkElement, 'via-link-element'],
+    [registerViaLinkHeader, 'via-link-header']
+  ];
+
+  // Testing creating registrations
+  for (const [registrationMethod, registrationMethodName] of registrationMethods) {
+    for (const updateViaCache of updateViaCacheValues) {
+      const testName = `register-${registrationMethodName}-updateViaCache-${updateViaCache}`;
+
+      promise_test(async () => {
+        await cleanup();
+
+        const opts = {scope};
+
+        if (updateViaCache) opts.updateViaCache = updateViaCache;
+
+        const reg = await registrationMethod(
+          `${scriptUrl}?test=${testName}`,
+          opts
+        );
+
+        const sw = reg.installing || reg.waiting || reg.active;
+        if (!sw) console.log(reg);
+        await waitForActivated(sw);
+        const values = await getScriptTimes(sw, testName);
+        await reg.update();
+
+        if (updateViaCache == 'all') {
+          assert_equals(reg.installing, null, "No new service worker");
+        }
+        else {
+          const newWorker = reg.installing;
+          assert_true(!!newWorker, "New worker installing");
+          const newValues = await getScriptTimes(newWorker, testName);
+
+          if (!updateViaCache || updateViaCache == 'imports') {
+            assert_not_equals(values.mainTime, newValues.mainTime, "Main script should have updated");
+            assert_equals(values.importTime, newValues.importTime, "Imported script should be the same");
+          }
+          else { // 'none'
+            assert_not_equals(values.mainTime, newValues.mainTime, "Main script should have updated");
+            assert_not_equals(values.importTime, newValues.importTime, "Imported script should have updated");
+          }
+        }
+
+        await cleanup();
+      }, testName);
+    }
+  }
+
+  // Testing changing registration
+  for (const updateViaCache1 of updateViaCacheValues) {
+    for (const updateViaCache2 of updateViaCacheValues) {
+      const testName = `register-with-updateViaCache-${updateViaCache1}-then-${updateViaCache2}`;
+
+      promise_test(async () => {
+        await cleanup();
+
+        const fullScriptUrl = `${scriptUrl}?test=${testName}`;
+        let opts = {scope};
+        if (updateViaCache1) opts.updateViaCache = updateViaCache1;
+
+        const reg = await navigator.serviceWorker.register(fullScriptUrl, opts);
+
+        const sw = reg.installing;
+        await waitForActivated(sw);
+        const values = await getScriptTimes(sw, testName);
+
+        opts = {scope};
+        if (updateViaCache2) opts.updateViaCache = updateViaCache2;
+
+        await navigator.serviceWorker.register(fullScriptUrl, opts);
+
+        // If there's no change, register should be a no-op.
```

Comment:
This was a bit subtle for me. Recommend saying "If there's no change to the updateViaCache value"